DfE.FindInformationAcademiesTrusts/Services/Export/Builders/AcademiesBuilder.cs

```diff
@@ -0,0 +1,131 @@
+using DfE.FindInformationAcademiesTrusts.Extensions;
+using DfE.FindInformationAcademiesTrusts.Services.Academy;
+
+namespace DfE.FindInformationAcademiesTrusts.Services.Export.Builders
+{
+    public class AcademiesBuilder(string sheetName = "Academies") : ExportBuilder(sheetName)
+    {
+        public AcademiesBuilder WriteTrustInformation(TrustSummary? trustSummary)
+        {
+            WriteTrustInformation<AcademiesBuilder>(trustSummary);
+            return this;
+        }
+
+        public AcademiesBuilder WriteHeaders()
+        {
+            var headers = new List<string>
+            {
+                ExportHelpers.SchoolName,
+                ExportHelpers.Urn,
+                ExportHelpers.LocalAuthority,
+                "Type",
+                "Rural or Urban",
+                ExportHelpers.DateJoined,
+                "Current Ofsted Rating",
+                ExportHelpers.BeforeOrAfterJoiningHeader,
+                "Date of Current Ofsted",
+                "Previous Ofsted Rating",
+                ExportHelpers.BeforeOrAfterJoiningHeader,
+                "Date of Previous Ofsted",
+                "Phase of Education",
+                ExportHelpers.AgeRange,
+                "Pupil Numbers",
+                "Capacity",
+                "% Full",
+                "Pupils eligible for Free School Meals",
+                "LA average pupils eligible for Free School Meals",
+                "National average pupils eligible for Free School Meals",
+                "Pupils eligible for Free School Meals"
+            };
+
+            WriteHeaders(headers);
+
+            return this;
+        }
+
+        public AcademiesBuilder WriteRows(AcademyDetails[] academies, AcademyOfsted[] academiesOfstedRatings,
+            AcademyPupilNumbers[] academiesPupilNumbers,
+            AcademyFreeSchoolMealsServiceModel[] academiesFreeSchoolMeals)
+        {
+            foreach (var details in academies)
+            {
+                var ofstedData = academiesOfstedRatings.Single(x => x.Urn == details.Urn);
+                var pupilData = academiesPupilNumbers.Single(x => x.Urn == details.Urn);
+                var freeMealsData = academiesFreeSchoolMeals.Single(x => x.Urn == details.Urn);
+
+                GenerateAcademyRow(details, ofstedData, pupilData, freeMealsData);
+                AddRow();
+            }
+
+            return this;
+        }
+
+        private void GenerateAcademyRow(
+            AcademyDetails academy,
+            AcademyOfsted ofstedData,
+            AcademyPupilNumbers pupilNumbersData,
+            AcademyFreeSchoolMealsServiceModel freeSchoolMealsData)
+        {
+            var previousRating = ofstedData.PreviousOfstedRating;
+            var currentRating = ofstedData.CurrentOfstedRating;
+            var percentageFull = ExportHelpers.CalculatePercentageFull(pupilNumbersData.NumberOfPupils, pupilNumbersData.SchoolCapacity);
```

Comment:
If we used the service objects rather than the repo objects then we wouldn't need to duplicate logic like `CalculatePercentageFull` which lives in the services